src/combinatorics/simple_graph/order.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2021 Eric. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Rodriguez
+-/
+import combinatorics.simple_graph.subgraph
+
+/-!
+# Order relation on simple graphs
+
+This file gives `simple_graph` a `bounded_lattice` instance under the `is_subgraph` relation, and
+proves a couple lemmata about how this relation interacts with `simple_graph.subgraph.spanning_coe`.
+
+# Main definitions:
+
+* `simple_graph.is_subgraph`: A `simple_graph` is a subgraph of another if its `adj` is dominated by
+  the other graph's.
+
+* `bounded_lattice` instance: With this definition, `simple_graph` forms a `bounded_lattice`.
+
+* `simple_graph.to_subgraph`: If a `simple_graph` is a subgraph of another, then you can turn it
+  into a member of the larger graph's `simple_graph.subgraph` type.
+
+-/
+
+namespace simple_graph
+
+variable {V : Type*}
+
+/-- The relation that one `simple_graph` is a subgraph of another. -/
+def is_subgraph (x y : simple_graph V) := ∀ {v w : V}, x.adj v w → y.adj v w
+
+/-- The union of two `simple_graph`s. -/
+def union (x y : simple_graph V) : simple_graph V :=
+{ adj := λ v w, x.adj v w ∨ y.adj v w,
+  sym := λ v w h, by rwa [x.adj_comm, y.adj_comm] }
+
+/-- The intersection of two `simple_graph`s. -/
+def inter (x y : simple_graph V) : simple_graph V :=
+{ adj := λ v w, x.adj v w ∧ y.adj v w }
+
+instance : has_union (simple_graph V) := ⟨union⟩
+instance : has_inter (simple_graph V) := ⟨inter⟩
+
+instance : bounded_lattice (simple_graph V) :=
+{ le := is_subgraph,
+  sup := union,
+  inf := inter,
+  top := complete_graph V,
+  bot := empty_graph V,
+  le_refl := λ _ _ _ h, h,
+  le_trans := λ x y z hxy hyz _ _ h, hyz (hxy h),
+  le_antisymm := λ x y hxy hyx, by { ext v w, exact ⟨λ h, hxy h, λ h, hyx h⟩ },
+  le_top := λ x v w h, x.ne_of_adj h,
+  bot_le := λ x v w h, h.elim,
+  sup_le := λ x y z hxy hyz v w h, h.cases_on (λ h, hxy h) (λ h, hyz h),
+  le_sup_left := λ x y v w h, or.inl h,
+  le_sup_right := λ x y v w h, or.inr h,
+  le_inf := λ x y z hxy hyz v w h, ⟨hxy h, hyz h⟩,
+  inf_le_left := λ x y v w h, h.1,
+  inf_le_right := λ x y v w h, h.2 }
```

Comment:
```suggestion
instance : bounded_lattice (simple_graph V) :=
{ le := is_subgraph,
  sup := union,
  inf := inter,
  top := complete_graph V,
  bot := empty_graph V,
  le_top := λ x v w h, x.ne_of_adj h,
  bot_le := λ x v w h, h.elim,
  sup_le := λ x y z hxy hyz v w h, h.cases_on (λ h, hxy h) (λ h, hyz h),
  le_sup_left := λ x y v w h, or.inl h,
  le_sup_right := λ x y v w h, or.inr h,
  le_inf := λ x y z hxy hyz v w h, ⟨hxy h, hyz h⟩,
  inf_le_left := λ x y v w h, h.1,
  inf_le_right := λ x y v w h, h.2,
  .. partial_order.lift simple_graph.adj (λ (x y : simple_graph V) h, by { ext, rw h }) }
```

I think a `bounded_lattice.lift` might be useful here.

Reply:
[Zulip](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there.20code.20for.20X.3F/topic/Lifting.20lattice.20structures/near/246215892)

Reply:
for sure, I find it very weird that with your example, removing `le` ruins it all, even though the `bounded_lattice` you create has the `lt` implied by `partial_order.lift` instead of the `lt` implied by `is_subgraph`

Reply:
also, i'll change this on the `subgraph` one too if we find a good way as it's pretty much a copy-paste of that

Reply:
It breaks if you remove `le` because `pi.le` is defined with `(v w : V)` not `⦃v w : V⦄`, so you need extra underscores. I think your custom binders are better though.

Reply:
(I am leaving this unresolved in case we get `bounded_lattice.lift` or further ideas)